prov/verbs/src/verbs_domain.c

```diff
@@ -256,6 +256,36 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
 	return ret;
 }
 
+static int fi_ibv_trywait(struct fid_fabric *fabric, struct fid **fids, int count)
+{
+	struct fi_ibv_cq *cq;
+	fi_ibv_trywait_func trywait;
+	int ret, i;
+
+	for (i = 0; i < count; i++) {
+		switch (fids[i]->fclass) {
+		case FI_CLASS_CQ:
+			cq = container_of(fids[i], struct fi_ibv_cq, cq_fid.fid);
+			trywait = cq->trywait;
+			break;
+		case FI_CLASS_EQ:
+			/* We are always ready to wait on an EQ since
+			 * rdmacm EQ is based on an fd */
+			continue;
+		case FI_CLASS_CNTR:
+		case FI_CLASS_WAIT:
+			return -FI_ENOSYS;
+		default:
+			return -FI_EINVAL;
+		}
+
+		ret = trywait(fids[i]);
+		if (ret)
+			return ret;
```

Comment:
I would drop the trywait variable and just call cq->trywait directly in the switch statement.